service/Core/Configuration/ServiceConfig.cs

```diff
@@ -30,18 +29,8 @@ public class ServiceConfig
     public Dictionary<string, HandlerConfig> Handlers { get; set; } = new();
 
     /// <summary>
-    /// The maximum allowed size in megabytes for a request body posted to the upload endpoint.
+    /// The maximum allowed size in bytes for a request body posted to the upload endpoint.
     /// If not set the solution defaults to 30,000,000 bytes (~28.6 MB) (ASP.NET default).
     /// </summary>
-    public long? MaxUploadSizeMb { get; set; } = null;
-
-    public long? GetMaxUploadSizeInBytes()
-    {
-        if (this.MaxUploadSizeMb.HasValue)
-        {
-            return Math.Min(10, this.MaxUploadSizeMb.Value) * 1024 * 1024;
```

Comment:
I think this is the only line that needs fixing, using `Math.Max`. I'd keep also the min of 10Mb in place.

Reply:
So don't you want to handle scenarios where, for example, I need to limit the upload to 1 MB?